cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -884,6 +898,111 @@ impl CoreEnvironment<ReadOnly> {
         self.replace_with(temp_env)?;
         Ok(store_path)
     }
+
+    /// Should not be called with
+    /// !migration_info.needs_manifest_migration && !migration_info.needs_upgrade
```

Comment:
Suggestion: this seems like something that should be prevented before ever calling this function e.g. make it impossible to construct this condition. For the sake of time let's ignore it though.

Reply:
I guess we could use a 3 state enum but that seems a little clunky. Do you have something else in mind?